### PR TITLE
Bug 1763025 - Extend /push/ endpoint to allow filtering out reviewbot revisions

### PR DESCRIPTION
**Pull request by beatrice-acasandrei**

### CODE REVIEW

File: treeherder/webapp/api/push.py [hunk 1]
```diff
@@ -159,6 +159,11 @@ def list(self, request, project):
             else:
                 pushes = pushes.filter(author=author)
 
+        hide_reviewbot_pushes = filter_params.get("hide_reviewbot_pushes")
+        exclude_reviewbot_author = hide_reviewbot_pushes == "true"
+        if exclude_reviewbot_author:
```
Comment (eng-esther) [inline treeherder/webapp/api/push.py:164]:
```suggestion
        if filter_params.get("hide_reviewbot_pushes") == "true":
```

**Review by eng-esther — APPROVED** [review]

File: tests/webapp/api/test_push_api.py [hunk 2]
```diff
@@ -300,6 +300,34 @@ def test_push_author(client, test_repository):
     assert set([result['id'] for result in results]) == set([1, 2])
 
 
+def test_push_reviewbot(client, test_repository):
```
Comment (alexandru-io) [inline tests/webapp/api/test_push_api.py:303]: could you try doing like [here](https://github.com/mozilla/treeherder/pull/7416/files#diff-9106a24b9166c2c3b076b01263212cb02b481cd2b146f31173015fdafb15adc5R191) ? using @pytest.mark.parameterize

**Comment by alexandru-io** [comment]
Try adding `hide_reviewbot_pushes` in the `if` at line 75. There should be added as key so the get should work.

**Comment by jmaher** [comment]
this python test failure is a test issue that seems to be time related- it is showing up on all pull requests now.  please don't spend time investigating this.

**Comment by jmaher** [comment]
if you rebase this revision the tests should pass again

**Pull request merged**